pkg/provider/aws/resources/elasticache.go

```diff
@@ -0,0 +1,91 @@
+package resources
+
+import (
+	"fmt"
+
+	"github.com/klothoplatform/klotho/pkg/config"
+	"github.com/klothoplatform/klotho/pkg/core"
+)
+
+type (
+	Elasticache struct {
+		Name            string
+		Engine          string
+		CloudwatchGroup *LogGroup
+		SubnetGroup     *ElasticacheSubnetgroup
+		SecurityGroups  []*SecurityGroup
+		ConstructsRef   []core.AnnotationKey
+	}
+
+	ElasticacheSubnetgroup struct {
+		Name          string
+		Subnets       []*Subnet
+		ConstructsRef []core.AnnotationKey
+	}
+)
+
+const (
+	EC_TYPE   = "elasticache"
+	ECSN_TYPE = "elasticache_subnetgroup"
+)
+
+// Provider returns name of the provider the resource is correlated to
+func (ec *Elasticache) Provider() string {
+	return AWS_PROVIDER
+}
+
+// KlothoResource returns AnnotationKey of the klotho resource the cloud resource is correlated to
+func (ec *Elasticache) KlothoConstructRef() []core.AnnotationKey {
+	return ec.ConstructsRef
+}
+
+// ID returns the id of the cloud resource
+func (ec *Elasticache) Id() string {
+	return fmt.Sprintf("%s:%s:%s", ec.Provider(), EC_TYPE, ec.Name)
+}
+
+// Provider returns name of the provider the resource is correlated to
+func (ecsn *ElasticacheSubnetgroup) Provider() string {
+	return AWS_PROVIDER
+}
+
+// KlothoResource returns AnnotationKey of the klotho resource the cloud resource is correlated to
+func (ecsn *ElasticacheSubnetgroup) KlothoConstructRef() []core.AnnotationKey {
+	return ecsn.ConstructsRef
+}
+
+// ID returns the id of the cloud resource
+func (ecsn *ElasticacheSubnetgroup) Id() string {
+	return fmt.Sprintf("%s:%s:%s", ecsn.Provider(), ECSN_TYPE, ecsn.Name)
+}
+
+func CreateElasticache(cfg *config.Application, dag *core.ResourceGraph, source core.Construct) *Elasticache {
```

Comment:
i would take in a list of constructs because multiple could depend on this and then we can set the reference to all of them. Something i started to realize with creating new resources